src/lib/mina.ts

```diff
@@ -389,12 +392,23 @@ const defaultNetworkConstants: NetworkConstants = {
 function LocalBlockchain({
   proofsEnabled = true,
   enforceTransactionLimits = true,
+  networkId = 'testnet' as NetworkId,
 } = {}) {
   const slotTime = 3 * 60 * 1000;
   const startTime = Date.now();
   const genesisTimestamp = UInt64.from(startTime);
   const ledger = Ledger.create();
   let networkState = defaultNetworkState();
+  let minaNetworkId: NetworkId = 'testnet';
+
+  if (networkId) {
+    if (networkId !== 'mainnet' && networkId !== 'testnet') {
+      throw Error(
+        "Invalid network id specified. Please use 'mainnet' or 'testnet'"
+      );
+    }
```

Comment:
remove this check - we consistently rely on types and the type forbids passing anything else than `"mainnet"` or `"testnet"`

Reply:
@mitschabaude 
The idea behind this check is to fail as soon as network is configured. Without this checks in-browser vanilla JS will allow configure network with any value:

```javascript
const remoteBlockchain = Mina.Network({
  networkId: 'whatever',
  mina: 'http://localhost:8080/graphql',
})
Mina.setActiveInstance(remoteBlockchain)
```

And will only throw later when `networkId` will actually be used.

Do you still want these checks to be removed?

Reply:
yes, I'd remove them. the reasoning is consistency -- there are a million places where we don't validate input types and that break in arbitrary/unhelpful ways when passed an input of the wrong type.

if we validate inputs here, why not anywhere else?

the decision to not validate inputs in general comes with the expectation that o1js is used from typescript. Passing 'whatever' as in your example would result in an error at _compile time_, i.e. before your check would ever be run. from that perspective, the check is dead code.

if you think we shouldn't rely on typescript and do detailed input validation despite type restrictions, then this should be brought forth as part of a general discussion that applies to the entire code base; not added in an ad-hoc way like here which is inconsistent with the rest of the code base.

Reply:
Removed, internal discussion started. Thank you.